### ISSUE TITLE
Invidious integration, with private and public options

### ISSUE THREAD

**Issue by MagnaCapax** [MagnaCapax/PMSS#52]
We should integrate https://invidious.io/ -- private youtube front-end.

Users can choose whether it's public or private just for them.
Access path for public ought to be /public-USERNAME/invidious
Private /user-USERNAME/invidious or /user-USERNAME/apps/invidious

Preconfigured completely.
Enforcing script for basic settings which could be showstoppers if changed needs to be done as well. (Akin to people changing deluge/qbit ports and then wondering why they don't work)

**Comment by MagnaCapax** [comment]
Checked the LinuxServer image index (fleet) and didn’t find an Invidious image listed as of 2026-02-01. That suggests fleet.linuxserver.io does not currently provide an Invidious container (unless it exists under a different name).

— Sampsa Pellervoinen 🌱

**Comment by MagnaCapax** [comment]
Follow-up (2026-02-01): LinuxServer’s official images list in their docs does not include Invidious, and the Fleet page now redirects to that list. So I still don’t see evidence of a LinuxServer-maintained Invidious image.

— Sampsa Pellervoinen 🌱

**Comment by MagnaCapax** [comment]
## Verification Report — FAILED ✗

**Mode:** sweep | **Server:** <hostname> (Debian 11, PMSS git/main@2026-04-03)

### What Failed

**Layer 1 — No implementation found.** Code audit on git/main shows no implementation of the feature described in this issue. The `complete-verify` label was applied to a proposal, not a completed feature.

### Investigation

Full code audit conducted on PMSS git/main@2026-04-03. Searched all affected scripts, templates, and library files for implementation evidence. No matching code paths, provisioning hooks, or configuration changes found. Issue body describes a design proposal only.

### Verdict

`complete-verify` label **removed**. This issue remains open as a feature request. When implementation is ready, re-apply `complete-verify` and re-run QA sweep.

---

Väinämöinen <noreply@pulsedmedia.com>
(The words exist; the code does not — yet)

**Comment by MagnaCapax** [comment]
🔓 UNLOCKED — QA sweep assessment complete.

**Result:** SKIPPED

No deployable PMSS code implementation found for the acceptance criteria in this issue. The issue remains open for development. `complete-verify` label will be removed — re-add when implementation is committed.

**Comment by MagnaCapax** [comment]
## Investigation Report

**Issue:** #52 — Invidious integration, with private and public options
**Investigated:** 2026-05-29
**Status:** PARTIAL (inert proxy/UI plumbing only; the feature does not function)

### Summary
The reverse-proxy + UI scaffolding for Invidious exists and is correctly designed, but there is no
installer, no Invidious config/database/companion, and — decisively — **Invidious cannot serve
YouTube from a datacenter IP in 2026** because YouTube blocks datacenter/VPN IP ranges. On a
blocked IP the Invidious container fails its token fetch and **restarts infinitely**, which would
burden a shared seedbox host. Recommendation: **SHOULD-NOT-IMPLEMENT as specified** (preconfigured,
working, fleet-wide), and record the decision so this issue stops being re-investigated.

### What The Issue Asks For
- Invidious deployed per user, with a user choice of **public** (`/public-USERNAME/invidious`) or
  **private** (`/user-USERNAME/invidious` or `/user-USERNAME/apps/invidious`).
- "Preconfigured completely" — works out of the box, usable by the user.
- An enforcement script for basic/showstopper settings (analogous to port-pinning for deluge/qbit).

### Implementation Status
Present today (matches the documented per-user lighttpd `custom.d` proxy pattern and ADR 0009):
- `scripts/lib/lighttpd/proxyFragments.php` — an `invidious` reverse-proxy fragment for both the
  `/public-<user>/invidious` and `/user-<user>/apps/invidious` paths the issue requested.
- `scripts/lib/lighttpd/userConfigApply.php` — reads `~/.invidiousPort`; if a valid port is pinned
  it writes the proxy fragment, otherwise it removes it.
- `etc/skel/www/index.php` — an Invidious tab **definition**. It only renders when a matching proxy
  fragment is present, so for a normal account (no port pinned) the tab does **not** appear.
- `scripts/lib/tests/development/invidiousLighttpdProxyTest.php` — a characterization test of the
  generated fragment + a source-string assertion. It verifies the plumbing, not a working app.
- `docs/contracts.md` — documents the conditional proxy behavior.

This is the accurate basis for the maintainer's 2026-04-01 note: it is **only** the proxy wiring.
Nothing in the normal flow ever sets `~/.invidiousPort`, so the wiring is inert — there is no
process listening behind it.

### What's Missing
- An installer (Invidious is not in the supported one-click container app set).
- Invidious `config.yml`, its PostgreSQL database, and the Invidious **companion** component.
- A working path past YouTube's datacenter-IP block (see below).
- The requested enforcement script.
- Anything that actually pins `~/.invidiousPort`.

### Root Blocker — upstream non-viability (verified against official docs)
PMSS hosts run on datacenter IPs. Invidious depends on YouTube, which blocks those ranges:
- Official docs: *"It is known that YouTube block datacenter and VPN IP addresses."*
  (https://docs.invidious.io/youtube-errors-explained/)
- The FAQ points instance operators to block-evasion guides — rotating an IPv6 address "for
  escaping YouTube blocking" and hiding "behind proxy servers (for escaping DMCA requests)."
  (https://docs.invidious.io/faq/)
- Invidious **Companion does not solve this**: it improves bot-detection resistance but *"cannot
  bypass YouTube's blanket IP-based blocking of datacenter addresses... on a VPS you need a proxy
  to a residential IP."* On a blocked IP the token fetch fails and a Docker deployment **restarts
  in a loop** (https://github.com/iv-org/invidious/issues/4978).

Bypassing the block requires routing YouTube requests through a **residential-classified** IP or
proxy, plus per-instance IPv6 rotation — fragile, cat-and-mouse measures the upstream project
itself says can take "months" to re-fix when YouTube changes. A shared datacenter seedbox host
cannot provide a residential egress safely or at fleet scale. This is **outside PMSS's control**.

### Mission Alignment
- **For:** Invidious is a privacy/information-freedom tool — strong alignment in *intent*.
- **Against (decisive):** A feature that cannot reliably work harms **reliability** (a crash-looping
  container degrades a shared host for other tenants) and **honesty** (shipping a "YouTube frontend"
  that shows "sign in to confirm you're not a bot" is a broken promise). Reliability-over-heroics
  argues against owning a feature that breaks every time a third party changes its defenses.

### Devil's Advocate
**Why NOT build:** It cannot work out of the box on datacenter IPs; the default failure mode is a
resource-draining crash loop; bypassing requires fragile per-instance residential-egress hacks; the
maintenance burden is perpetual (YouTube repeatedly breaks Invidious).
**Could a lighter "per-user choice / one-click installer, BYO proxy" version count?** No — the issue
explicitly asks for "preconfigured completely" and "usable," and the 2026-04-01 note rejected the
scaffolding-only approach. A one-click installer that still requires the user to supply a residential
proxy is the opposite of "preconfigured," and it still crash-loops by default.
**Rebuttal in favor?** The only honest rebuttal is "do it when upstream viability returns or a
sustainable residential egress exists" — i.e., defer, not build now.

### Anti-Pattern Hunt
- REGEX_FOR_SEMANTICS: PASS — proxy fragments use regex for URL routing (legitimate), no semantic scoring.
- PREMATURE_OPTIMIZATION: PASS — recommendation is to not build; no speculative optimization proposed.
- FALSE_COMPLETION_CLAIM: PASS — reconciled: requested scope (working, preconfigured Invidious) vs
  delivered evidence (inert proxy wiring only). This issue's history is itself a false-completion case.
- OPERATOR_HANDOFF_ANTIPATTERN: PASS — a definite recommendation (SHOULD-NOT-IMPLEMENT) is made, not deferred.
- FABRICATED_AUTHORITY_OR_PATH: PASS — every cited file/ADR was read this run; external claims cite official docs.
- FAIL_OPEN_SAFETY_BEHAVIOR: PASS — N/A; no safety gate changed. (Note: the crash-loop is an upstream fail-*open* hazard, an argument against deploying.)
- DRY_RUN_THEATER: PASS — N/A.
- INVESTIGATION_LOOP_THEATER: PASS — prior ~10 QA cycles only re-confirmed "no implementation"; this pass answers the unasked root question (upstream viability), which ends the loop rather than extends it.

### Related Incidents & Data
- ~10 QA verification cycles (2026-04-01 → 2026-04-10) all SKIPPED/FAILED; `complete-verify`
  repeatedly applied to a proposal and removed. High churn, zero progress — the cost of not asking
  the root viability question.
- The supported one-click container apps (jellyfin, qbittorrent, *arr, etc.) are local-content or
  self-contained; none depend on an actively-blocking third party, which is why they work and
  Invidious would not.

### Cross-Issue Analysis
Searched the open-issue list for shared component/files/scope. No duplicate or superseding issue.
Foundational/related (not merges):
- **#10** (dynamic deluge/qbittorrent "app launchers") and **#18** (LinuxServer.io containers) —
  the app-install pattern Invidious would have to follow.
- **#559** (move *arr app management to per-account) — the per-account app-management direction an
  Invidious install would belong to.
- **#326 / ADR 0015** (browser console deferred until packaged) — the closest *precedent*: a wanted
  feature deferred rather than shipped as a placeholder.
No cross-issue file conflicts detected.

### Dependencies & Blockers
Hard external blocker: YouTube's datacenter-IP blocking + the requirement for residential egress.
This is not resolvable inside PMSS. Until upstream viability changes (or a sustainable, isolated
residential egress is engineered and reviewed), the feature's core function cannot be delivered.

### Reuse, Refactor & Test Strategy
- **Existing reuse points:** The per-user lighttpd `custom.d` reverse-proxy pattern (ADR 0009) is
  already the supported, working way to expose any user-run Docker web app over HTTPS — a customer
  has already self-deployed an unrelated Docker web app this way with no PMSS preconfiguration. The
  Invidious proxy fragment already reuses this correctly.
- **Refactor opportunities:** None worth doing for this feature — building a bespoke Invidious
  installer to host a non-functional app would be wasted work. If anything, the *opposite*:
  the inert Invidious placeholder (proxy fragment + tab + test) should be removed or gated, since
  ADR 0015 explicitly rejects shipping placeholder dead UX.
- **Test categories (only if ever built):** would require a real end-to-end "does a video actually
  play from this host's egress" check — which today would fail by design. That failing E2E is the
  evidence that the feature should not ship now.
- **Minimum-code-for-correctness:** The smallest correct action is **no new code** — decline, record
  the decision, and clean up the placeholder. "The best part is no part."
- **Mission/business success signal:** A truthful success signal cannot be named while the feature
  cannot serve a single working video from a PMSS host. There is no retention/revenue signal for a
  feature that crash-loops on deploy and generates support load. Per the prompt's own test, that
  makes a build today decorative → SHOULD-NOT-IMPLEMENT.

### Recommendation
**SHOULD-NOT-IMPLEMENT** (as specified). The 2024 premise — that Invidious works as a
preconfigured YouTube frontend — is no longer true in 2026 for datacenter-hosted instances. The
proxy scaffolding is sound but inert; completing the installer would ship a feature that fails by
default (and crash-loops), creating support burden and a broken-promise UX. The honest path is to
**defer until upstream viability returns or a sustainable, isolated residential egress is designed
and security-reviewed**, mirroring the defer-until-ready ADR pattern (0011/0012/0013/0015).

If the maintainer wants to proceed despite this with a fragile, BYO-residential-proxy variant, that
is a conscious business trade-off to make **with this information in hand** — but the grounded
engineering recommendation, given the verified data, is do-not-build-now.

### Additive Recommendation Flags
- **RECOMMEND_ADR** — Draft title: *"Decline/defer preconfigured Invidious integration —
  datacenter-IP non-viability."* Context: Invidious depends on YouTube, which blocks datacenter
  IPs; Companion cannot bypass blanket IP blocking on a VPS; default deploy crash-loops; the proxy
  scaffolding already exists but the feature cannot function without residential egress. Options
  considered: build installer now (rejected — ships a broken/crash-looping feature); BYO-proxy
  one-click (rejected — not "preconfigured," still crash-loops by default); defer until upstream
  viability or an isolated residential egress is engineered + reviewed (chosen). Consequence:
  recording this stops the recurring re-investigation of #52. Mirrors ADR 0015's reasoning.
- **RECOMMEND_NEW_ISSUE** — Title: *"Remove or gate the inert Invidious proxy/UI placeholder."*
  Scope: per ADR 0015 (no placeholder dead UX), remove or feature-gate the dormant Invidious
  fragment/tab/test; mind the edge case of any account that manually pinned `~/.invidiousPort`
  (don't break it).

---
Väinämöinen <noreply@pulsedmedia.com>
*First the origin; then the cure. The words exist; the upstream does not — yet.*

**Comment by MagnaCapax** [comment]
## Investigation Report

**Issue:** #52 — Invidious integration, with private and public options
**Investigated:** 2026-06-01
**Status:** PARTIAL (inert proxy/UI plumbing only; feature non-functional)

> This pass **independently re-verifies and confirms** the thorough 2026-05-29 report above rather than
> re-deriving it. New contributions: source-level re-verification of every decisive claim, and a note on
> why this issue keeps reappearing (with the fix). The open question on #52 is no longer technical.

### Summary
The reverse-proxy + UI scaffolding for Invidious exists and is correctly designed, but it is **inert**
(nothing ever activates it); there is no installer/config/database/companion/enforcement script; and
self-hosted Invidious is **not viable from datacenter IPs in 2026** (YouTube blocks datacenter ranges,
and a blocked instance crash-loops). Recommendation is unchanged: **SHOULD-NOT-IMPLEMENT as specified.**

### What The Issue Asks For
Per-user Invidious with a **public** (`/public-USERNAME/invidious`) or **private**
(`/user-USERNAME/invidious` / `/user-USERNAME/apps/invidious`) choice, "preconfigured completely" (works
out of the box), plus an enforcement script for showstopper settings (like the deluge/qbit port pinning).

### Implementation Status
Verified from source this run:
- `scripts/lib/lighttpd/proxyFragments.php:61-67` — an `invidious` reverse-proxy fragment for both the
  public and private paths the issue requested.
- `scripts/lib/lighttpd/userConfigApply.php:71-76` — reads `~/.invidiousPort`; writes the proxy fragment
  if a valid port is pinned, otherwise removes it.
- `etc/skel/www/index.php:180-183` — an Invidious tab **definition**; it renders only when a matching proxy
  fragment is present (`pmssLocalFrameProxyAppFramesRead`), so for a normal account it does **not** appear.
- `scripts/lib/tests/development/invidiousLighttpdProxyTest.php` — a characterization test of the generated
  fragment. It tests the plumbing, not a working app.
- `docs/contracts.md:440` — documents the conditional behaviour.

**Decisive verification:** a whole-repository search confirms **nothing ever writes `~/.invidiousPort`**. The
trigger is never pulled, so the proxy is never written and the tab never shows. This is exactly the basis for
the maintainer's 2026-04-01 note — it is *only* the proxy wiring. The placement itself is correct per ADR 0009
(per-app proxying lives in per-user lighttpd, not fleet nginx).

### What's Missing
An installer; Invidious `config.yml`; its PostgreSQL database; the Invidious companion; a working YouTube-egress
path; the enforcement script; anything that sets `~/.invidiousPort`. The proxy is the easy ~5%; the hard ~95%
(install + egress) is absent.

### Mission Alignment
- **For:** Invidious is a privacy / information-freedom tool — strong alignment in intent.
- **Against (decisive):** a feature that cannot reliably work harms **reliability** (a crash-looping container
  degrades a shared host for other tenants) and **honesty** (shipping a "YouTube frontend" that shows "sign in
  to confirm you're not a bot" is a broken promise). Reliability-over-heroics argues against owning a feature
  that breaks whenever a third party changes its defences.

### Devil's Advocate
**Why NOT build:** it cannot work out of the box on datacenter IPs; the default failure mode is a resource-draining
crash loop on a shared host; bypassing the block requires a residential egress not available for customers at fleet
scale; the maintenance burden is perpetual (YouTube repeatedly breaks Invidious).
**Could a lighter "one-click installer, bring-your-own residential proxy" variant count?** It would not satisfy
"preconfigured completely / usable," and it still crash-loops by default — the opposite of what the issue asks.
**Honest rebuttal in favour of building:** only "do it when upstream viability returns, or once a sustainable,
isolated residential egress is engineered and reviewed" — i.e. defer, not build now.

### Anti-Pattern Hunt
- REGEX_FOR_SEMANTICS: PASS — proxy fragments use regex for URL routing (legitimate); no semantic scoring introduced.
- PREMATURE_OPTIMIZATION: PASS — recommendation is do-not-build; no speculative optimization proposed.
- FALSE_COMPLETION_CLAIM: PASS — requested scope (working, preconfigured Invidious) reconciled against delivered
  evidence (inert proxy wiring only, no `.invidiousPort` writer anywhere). This issue's history *is* the textbook case.
- OPERATOR_HANDOFF_ANTIPATTERN: PASS — a definite engineering recommendation is given (SHOULD-NOT-IMPLEMENT), not deferred.
- FABRICATED_AUTHORITY_OR_PATH: PASS — every cited file/line/ADR was opened this run; external claims rest on official docs.
- FAIL_OPEN_SAFETY_BEHAVIOR: PASS — N/A (no safety gate changed). The upstream crash-loop is itself an argument against deploying.
- DRY_RUN_THEATER: PASS — N/A.
- INVESTIGATION_LOOP_THEATER: **FLAGGED on the issue's history, not on this verdict** — ~10 verification cycles plus a deep
  2026-05-29 investigation plus this run all reach the same answer. The fix is to record and close, not to investigate again.

### Related Incidents & Data
- ~10 verification cycles (2026-04-01 → 2026-04-10) all SKIPPED/FAILED; `complete-verify` repeatedly applied to a proposal
  and removed. High churn, zero progress — the cost of not asking the root viability question.
- Datacenter-IP non-viability is corroborated by two independent internal research notes (2026-05-26 and 2026-05-29), each
  citing official Invidious docs and upstream issue trackers.
- The supported one-click apps (jellyfin, qbittorrent, *arr, etc.) are local-content or self-contained; none depends on an
  actively-blocking third party — which is why they work and Invidious would not.

### Cross-Issue Analysis
No duplicate or superseding issue for #52. Relationships:
- **#378** (`agentic-issues.sh: exclude should-not-implement from candidate pool`) — the process improvement that would stop a
  declined issue like #52 from being repeatedly re-triaged. Closest thing to a blocker for ending the cycle.
- **#18** (LinuxServer.io containers), **#57** (remote desktop container), **#171** (container runtime options for user isolation)
  — the container-app integration pattern Invidious would follow; none overlaps #52's files.
- **#470** (per-user `rtorrentEnabled` toggle) / **#37** (user settings: disable software) — same per-user app-tab/toggle subsystem
  (`etc/skel/www/index.php`, `userConfigApply.php`); landing awareness only, no conflict.
- ADR 0015 / issue #326 (browser console) — the precedent: a wanted feature deferred rather than shipped as a placeholder.
No cross-issue file conflicts detected.

### Dependencies & Blockers
Hard external blocker: YouTube's datacenter-IP blocking plus the requirement for a residential egress. No customer-facing
residential-IP egress is available at fleet scale, and providing one is out of scope here. Not resolvable inside PMSS until
upstream viability changes.

### Reuse, Refactor & Test Strategy
- **Existing reuse points:** the per-user lighttpd `custom.d` reverse-proxy pattern (ADR 0009) already exposes any user-run
  Docker web app over HTTPS — a customer can self-deploy Invidious today with zero PMSS-specific code if they solve their own
  egress. The Invidious fragment already reuses this correctly.
- **Refactor opportunities:** none worth doing — building a bespoke installer to host a non-functional app is wasted work.
  The dormant Invidious placeholder could optionally be removed or documented as opt-in-only (low priority — it is not
  user-visible dead UX, so ADR 0015's "no dead UX" rule does not compel removal).
- **Test categories (only if ever built):** a real end-to-end "does a video actually play from this host's egress" check —
  which today fails by design. That failing E2E is itself the evidence not to ship.
- **Minimum-code-for-correctness:** the smallest correct action is **no new code** — decline, record the decision, close.
- **Mission/business success signal:** none can be named truthfully while the feature cannot serve a single working video
  from a PMSS host, and it would generate support load + crash-loop risk → a build today would be decorative.

### First-Principles / Architectural Note
Question the requirement: "preconfigured completely / works out of the box" rested on the 2024-era assumption that Invidious
works as a hosted YouTube frontend on any IP. That assumption is **false in 2026** for datacenter-hosted instances — so the
requirement is unsatisfiable as written. The "best part is no part" path wins: drop the bespoke-installer requirement; the
generic Docker-web-app + lighttpd `custom.d` pattern already covers the only viable case (a user who solves their own egress).
Recommendation EXTENDS the defer-rather-than-ship-a-placeholder logic of ADR 0015 (and is a stronger decline, since this is
upstream non-viability, not merely a packaging gap) and ADHERES to ADR 0009 for proxy placement.

### Recommendation
**SHOULD-NOT-IMPLEMENT (as specified).** The 2024 premise — that Invidious works as a preconfigured YouTube frontend — is no
longer true in 2026 for datacenter-hosted instances. The proxy scaffolding is sound but inert; completing the installer would
ship a feature that fails by default (and crash-loops on a shared host), creating support burden and a broken-promise UX.

The reason this issue keeps reappearing is **not** a lack of investigation — it is that the verdict was never recorded and the
issue never closed. The constructive resolution: (1) record an ADR for the decline, and (2) close #52 (or consciously override
to a bring-your-own-residential-proxy variant with this information in hand). #378 tracks the process change that prevents
declined issues from re-triaging. Not auto-closed here because this is a non-viability decline (not a supersession) on an
operator-authored issue — the close/override is the maintainer's call.

### Additive Recommendation Flags
- **RECOMMEND_ADR** — Draft title: *"Decline/defer preconfigured Invidious integration — datacenter-IP non-viability."*
  Context: Invidious depends on YouTube, which blocks datacenter IPs; the companion cannot bypass blanket IP blocking; default
  deploy crash-loops; the proxy scaffolding exists but the feature cannot function without a residential egress not available
  for customers at fleet scale. Options considered: build the installer now (rejected — ships a broken/crash-looping feature);
  bring-your-own-proxy one-click (rejected — not "preconfigured," still crash-loops by default); defer until upstream viability
  or an isolated residential egress is engineered + reviewed (chosen). Consequence: recording this stops the recurring
  re-investigation of #52. Mirrors ADR 0015.

---
Väinämöinen <noreply@pulsedmedia.com>
*First the origin; then the cure. The words exist; the upstream does not — yet.*

**Comment by MagnaCapax** [comment]
## Investigation Report — Loop-Termination Confirmation (pass 3)

**Issue:** #52 — Invidious integration, with private and public options
**Investigated:** 2026-06-02
**Status:** PARTIAL (inert proxy/UI plumbing only; the feature does not function)

> **This is the third independent investigation to reach the same verdict** (after the
> 2026-05-29 and 2026-06-01 reports already in this thread). Rather than re-derive the
> technical case a third time — which would itself be investigation-loop theater — this pass
> (a) re-verifies the one decisive fact against current `main`, (b) confirms nothing changed
> since 2026-06-01, and (c) escalates the **only remaining action**, which is a maintainer
> decision, not more investigation.

### TL;DR — what actually needs to happen
The engineering question is settled: **SHOULD-NOT-IMPLEMENT as specified.** The loop persists
only because the verdict was never *recorded* and the issue was never *closed/relabeled*, so the
agentic candidate selector keeps re-picking it. Two maintainer actions end it for good:
1. **Record an ADR** ("Decline/defer preconfigured Invidious — datacenter-IP non-viability"),
   mirroring the defer-until-viable ADR pattern (0011/0012/0013/0015). Draft text below.
2. **Close this issue** (as won't-do) **or relabel it deferred** — maintainer's call, since you
   authored it. **#378** ("exclude should-not-implement from candidate pool") is the systemic fix
   that stops the re-selection regardless.

### What The Issue Asks For
Per-user Invidious with a **public** (`/public-USERNAME/invidious`) or **private**
(`/user-USERNAME/invidious` / `/user-USERNAME/apps/invidious`) choice, "preconfigured completely"
(works out of the box), plus an enforcement script for showstopper settings (like deluge/qbit port
pinning).

### Implementation Status — re-verified against current `main` (HEAD 2026-06-01)
The reverse-proxy + UI scaffolding exists and is correctly *placed*, but it is **inert**:
- `scripts/lib/lighttpd/proxyFragments.php` — defines the `invidious` reverse-proxy fragment for
  both requested paths.
- `scripts/lib/lighttpd/userConfigApply.php:71` — **reads** `~/.invidiousPort`; only if a valid
  port (1024–65535) is pinned does it write the proxy fragment, otherwise it removes it.
- `etc/skel/www/index.php` — an Invidious tab **definition** that renders only when a matching
  fragment exists, so for a normal account it does **not** appear (dormant, not user-visible dead UX).
- `scripts/lib/tests/development/invidiousLighttpdProxyTest.php` — characterization test of the
  generated fragment (tests the plumbing, not a working app).

**Decisive re-check this pass:** a whole-tree writer search confirms **nothing in PMSS ever writes
`~/.invidiousPort`** — the trigger is never pulled. Recent commits (proxy-port refactors, user-config
decompose, panel-tab offset fix) carried this dead branch along but did not change its inert nature.
This is exactly the basis for the 2026-04-01 maintainer note: it is *only* the proxy wiring.

### What's Missing
An installer; Invidious `config.yml`; its PostgreSQL database; the Invidious companion; a working
YouTube-egress path; the enforcement script; anything that sets `~/.invidiousPort`. The proxy is the
easy ~5%; the hard ~95% (install + egress) is absent.

### Root Blocker (unchanged, now cross-confirmed by a 2nd independent source)
PMSS hosts run on datacenter IPs; YouTube blocks those ranges. Self-hosted Invidious from a
datacenter IP returns "sign in to confirm you're not a bot," and a blocked instance **crash-loops**
on its token fetch — a resource drain on a *shared* host. Invidious Companion does not bypass blanket
IP blocking on a VPS (it needs a residential proxy). PMSS operates **no customer-facing residential
egress pool**, and the block cannot be safely worked around at fleet scale. Public Invidious
instances are down to a handful in 2026. This is upstream and outside PMSS's control.

### Mission Alignment
- **For:** Invidious is a privacy / information-freedom tool — strong alignment in intent.
- **Against (decisive):** a feature that cannot reliably work harms **reliability** (crash-looping
  container degrades a shared host) and **honesty** (shipping a "YouTube frontend" that shows a bot
  wall is a broken promise). Reliability-over-heroics argues against owning a feature that breaks
  whenever a third party changes its defenses.

### Devil's Advocate
**Why NOT build:** cannot work out of the box on datacenter IPs; default failure is a crash loop;
bypassing requires fragile per-instance residential-egress hacks; perpetual maintenance burden.
**Does a lighter "one-click installer, BYO proxy" variant count?** No — the issue asks for
"preconfigured completely," the 2026-04-01 note rejected scaffolding-only, and a BYO-proxy variant
still crash-loops by default and is the opposite of "preconfigured."
**Honest rebuttal in favor?** Only "do it when upstream viability returns" — i.e. defer, not build now.

### Anti-Pattern Hunt
- REGEX_FOR_SEMANTICS: PASS — proxy fragments use regex for URL routing (legitimate); no semantic scoring.
- PREMATURE_OPTIMIZATION: PASS — recommendation is *not to build*; no speculative optimization proposed.
- FALSE_COMPLETION_CLAIM: PASS — requested scope (working, preconfigured Invidious) vs delivered
  evidence (inert proxy wiring only) reconciled; this issue's own history is the textbook false-completion case.
- OPERATOR_HANDOFF_ANTIPATTERN: PASS — a definite engineering verdict is given (SHOULD-NOT-IMPLEMENT);
  the only deferral is the *close decision*, which is genuinely maintainer-reserved (you authored it),
  and it ships with ready-to-apply ADR text + a two-option decision, not a bare "please decide."
- FABRICATED_AUTHORITY_OR_PATH: PASS — every cited file/ADR/issue was read this run; the inert-wiring
  claim was re-verified from current `main`.
- FAIL_OPEN_SAFETY_BEHAVIOR: PASS — no safety gate changed. (The upstream crash-loop is itself a
  fail-open hazard — an argument against deploying.)
- DRY_RUN_THEATER: PASS — N/A.
- INVESTIGATION_LOOP_THEATER: **PASS (actively avoided)** — this pass does not re-run QA or re-derive
  the technical case; it re-verifies one decisive fact, confirms zero state-delta since 2026-06-01,
  and converts the thread from "investigate again" to "record + decide." Further auto-investigation
  of #52 adds nothing until the ADR is recorded and the issue is closed/relabeled.

### Related Incidents & Data
- ~10 QA verification cycles (2026-04-01 → 2026-04-10) all SKIPPED/FAILED — `complete-verify`
  repeatedly applied to a proposal and removed. High churn, zero progress: the cost of not asking the
  root viability question.
- Two prior full investigations (2026-05-29, 2026-06-01) in this thread, plus this one — same verdict
  each time. Each re-pick is wasted compute on a decided question. Frequency-of-recurrence is itself
  the signal that the fix is structural (#378), not more analysis.
- The supported one-click container apps (jellyfin, qbittorrent, *arr, etc.) are local-content or
  self-contained; none depend on an actively-blocking third party — which is why they work and
  Invidious would not.

### Cross-Issue Analysis
Scanned the open-issue list for shared component/files/scope. **No duplicate or superseding issue.**
- **#378** (exclude should-not-implement from candidate pool) — **systemic loop fix**, still OPEN. It
  does *not* subsume #52 (it's a selector fix, not an Invidious successor), so it is not grounds to
  auto-close #52.
- Adjacent (not merges): **#18** (LinuxServer.io containers) and **#10** (app launchers) — the
  app-install pattern Invidious would have to follow; **ADR 0015 / #326** (browser console deferred) —
  the closest precedent (wanted feature deferred rather than shipped as a placeholder).
No cross-issue file conflicts detected.

### Dependencies & Blockers
Hard external blocker: YouTube's datacenter-IP blocking + the requirement for residential egress —
not resolvable inside PMSS. The loop-termination depends on a maintainer close/relabel decision and,
systemically, on #378 landing.

### Reuse, Refactor & Test Strategy
- **Existing reuse points:** the per-user lighttpd `custom.d` reverse-proxy pattern (ADR 0009) is
  already the supported way to expose any user-run Docker web app over HTTPS; the Invidious fragment
  reuses it correctly. A capable user who solves their own egress can already run Invidious this way
  with zero PMSS-specific work — so the bespoke preconfigured installer is the part that shouldn't exist.
- **Refactor opportunities:** none worth doing *for* this feature. The only code action with positive
  value is the *opposite* — eventually removing/gating the inert placeholder (LOW priority; it is
  dormant and harmless, and ADR 0015 only forbids *user-visible* dead UX).
- **Test categories (only if ever built):** a real end-to-end "does a video actually play from this
  host's egress" check — which today fails by design. That failing E2E *is* the evidence not to ship.
- **Minimum-code-for-correctness:** **no new code.** "The best part is no part." Record the decision,
  decide close/defer.
- **Mission/business success signal:** none can be named truthfully while the feature cannot serve a
  single working video from a PMSS host; a crash-looping deploy generates support load, not retention.
  Per the prompt's own test, that makes a build today decorative → SHOULD-NOT-IMPLEMENT.

### Deep Research Gate
SKIPPED: prior-investigation-count (3 prior passes incl. a 2026-05-29 deep investigation) + memory-citation
(datacenter-IP non-viability already established and cross-confirmed by two independent internal sources).
Further deep research is investigation-loop theater per this prompt's own skip criteria.

### Architectural Minimization
- **Step A — dumbest requirement:** "Invidious, preconfigured completely, fleet-wide." Written 2024-02-15
  on the then-true premise that self-hosted Invidious works. WHO: the maintainer, on a 2024 assumption.
  The underlying constraint (give users a privacy YouTube frontend) is real; the *mechanism* (self-host
  Invidious on datacenter IPs) became non-viable in 2024–2026. The requirement's premise is the failure.
- **Step B — alternatives:**
  - DELETE: (1) remove the inert proxy fragment + tab + test → eliminates the "show and pretend" surface;
    LOW priority (dormant, harmless, doubles as an opt-in hook). (2) delete the *requirement* (decline the
    feature) → this is the recommended path. **DELETE/NULL wins.**
  - MERGE: (1) fold any future "user-run web app" need into the generic Docker + lighttpd `custom.d`
    pattern (already exists per ADR 0009 + the Kiwix precedent) — makes a bespoke Invidious installer
    redundant. (2) merge into #18/#10 app-install work — rejected: those don't solve the egress blocker,
    so merging just relocates a non-viable feature.
  - REFACTOR: (1) generalize the existing per-user proxy catalog to add apps without bespoke code — it
    already does this; no refactor unlocks Invidious because the blocker is egress, not plumbing.
    (2) generalize the installer set — rejected, same reason.
  - NULL: if we don't do this, nothing breaks; users wanting Invidious can BYO-egress via the generic
    pattern. Concrete consequence of *building* it instead: crash-loop on deploy harms shared-host tenants.
- **Step C:** alternative wins — **no minimal design proposed because the DELETE/NULL alternative
  (decline the feature; optionally remove the dormant placeholder later) is the correct path.**

### ADR Adherence
- **EXTENDS** ADR 0015 (defer browser console until packaged + reviewed) — same "defer-until-viable, no
  placeholder dead UX" reasoning; the Invidious tab+proxy is the analogous placeholder.
- **ADHERES** ADR 0009 (per-app reverse proxy lives in per-user lighttpd) — the proxy placement is correct.
- **No Invidious ADR exists** (latest recorded is 0020) — hence RECOMMEND_ADR below; recording it is the
  durable artifact that ends the loop.

### Final AQFM Sweep
Question asked (verbatim): *"Issue #52 (MagnaCapax/PMSS): Invidious integration… Draft conclusion: 3rd
investigation; all reach SHOULD-NOT-IMPLEMENT because YouTube blocks datacenter IPs since 2024 … loop
persists because no ADR + issue never closed; #378 is the systemic fix. Draft recommendation:
SHOULD-NOT-IMPLEMENT + RECOMMEND_ADR, do not auto-close (non-viability not supersession). What
contradicting evidence … could revise this draft?"*
Sources returned: the synthesis retrieved only the issue's own cache entry (not the prior lessons read
directly this run), so it surfaced no contradiction. **Verdict: HELD** on primary sources read directly
this pass (the two prior in-thread reports + current-`main` source re-verification + a second independent
internal non-viability source).

### Adversarial Self-Challenge (disproving searches against own findings)
**Strongest finding 1**: Invidious is non-viable on PMSS datacenter IPs (crash-loops; needs residential egress).
- Disproving search 1: searched memory for any evidence Invidious works on datacenter IPs / block lifted →
  no counter-evidence; instead found confirming residential-proxy material.
- Disproving search 2 (broader, deep-memory): the 2026-05-26 YouTube agentic-landscape research →
  independently tables "Self-hosted Invidious | FAIL | Google blocks at IP level since 2024." 
- Outcome: **held** (now two independent internal sources + official docs from prior pass).

**Strongest finding 2**: the in-repo plumbing is inert (nothing writes `~/.invidiousPort`).
- Disproving search: whole-tree writer search against current `main` → only a *reader*
  (`userConfigApply.php:71`) found; no writer.
- Outcome: **held** (re-verified from live source this pass, not relayed).

**Least-evidenced finding** (lowest citation count): "the loop persists because no ADR + issue stays open."
- Disproving search: looked for counter-evidence that re-investigation is productive / an ADR wouldn't help →
  found the opposite: a documented cluster of analogous re-spawn loops where the disciplined response on
  zero-state-delta is to stop re-deriving and fix the selector. ADR absence is confirmed (latest is 0020).
- Outcome: **held, strengthened** by the analogous-loop precedent.

### Recommendation
**SHOULD-NOT-IMPLEMENT** (as specified) — re-confirmed for the 3rd time and re-verified from current
`main`. The 2024 premise (Invidious works as a preconfigured YouTube frontend) is false in 2026 for
datacenter-hosted instances; the default deploy crash-loops on a shared host; there is no safe fleet-scale
residential egress. The honest path is **defer until upstream viability returns** (or record as won't-do) —
not build now.

### Additive Recommendation Flags
- **RECOMMEND_ADR** — Draft for `docs/adr/0021-decline-defer-invidious-datacenter-ip-non-viability.md`:
  > **Title:** Decline/defer preconfigured Invidious integration — datacenter-IP non-viability
  > **Status:** Accepted
  > **Context:** #52 requested per-user, preconfigured Invidious (public/private). Invidious depends on
  > YouTube, which has blocked datacenter/VPN IP ranges since 2024; PMSS hosts are all datacenter IPs.
  > A blocked instance crash-loops on token fetch, draining a shared host. Companion cannot bypass blanket
  > IP blocking on a VPS without residential egress, which PMSS does not offer to customers at fleet scale.
  > Only inert proxy/UI plumbing exists (nothing sets `~/.invidiousPort`).
  > **Options considered:** (1) build the installer now — rejected, ships a crash-looping feature;
  > (2) BYO-residential-proxy one-click — rejected, not "preconfigured," still crash-loops by default;
  > (3) defer until upstream viability returns or an isolated residential egress is engineered + reviewed —
  > **chosen**. Users wanting it today can self-host via the generic per-user Docker + lighttpd proxy pattern (ADR 0009).
  > **Consequences:** Recording this stops the recurring re-investigation of #52; the dormant placeholder may
  > be removed/gated later (LOW priority; ADR 0015 forbids only *user-visible* dead UX). Mirrors ADR 0011/0012/0013/0015.

### Auto-Close Decision
AUTO_CLOSED_BLOCKED: trigger-not-met; reason=rationale-not-supersession.
SHOULD-NOT-IMPLEMENT here is upstream *non-viability*, not supersession; #378 is a selector fix that does not
subsume #52; you authored the issue and may wish to keep it open as explicitly *deferred*. Per standing
policy, non-supersession declines stay maintainer-visible — the close/relabel call is yours, and I will not
make it unilaterally despite the loop cost.

---
Väinämöinen <noreply@pulsedmedia.com>
*First the origin; then the cure. The words exist; the upstream does not — yet. This thread needs a decision, not a fourth investigation.*

**Comment by MagnaCapax** [comment]
## Investigation Report

**Issue:** #52 — Invidious integration, with private and public options
**Investigated:** 2026-06-02
**Status:** PARTIAL (inert proxy/UI plumbing only — the feature does not function)

> This is a re-verification of a **converged** issue (multiple prior passes, incl. the thorough
> 2026-05-29 report and a 2026-06-01 internal re-confirmation), not a fresh investigation. Posting
> it to (a) independently re-confirm the verdict from current `main`, (b) sharpen the reasoning of
> record, and (c) make the loop-termination action crisp. **This issue does not need another
> investigation — it needs a decision.**

### Summary
The reverse-proxy + tab + test scaffolding for Invidious exists and is correctly designed, but it is
**inert** (nothing pins the port; there is no installer). Invidious depends on YouTube, which blocks
datacenter IP ranges; serving it per-user would require residential-classified egress that is not
available for customer video streaming at fleet scale, and a blocked instance crash-loops on a shared
host. Verdict: **SHOULD-NOT-IMPLEMENT as specified** (preconfigured, working, fleet-wide).

### What The Issue Asks For
- Per-user Invidious, user choice of **public** (`/public-USERNAME/invidious`) or **private**
  (`/user-USERNAME/invidious` / `/user-USERNAME/apps/invidious`).
- "Preconfigured completely" — usable out of the box.
- An enforcement script pinning showstopper settings (analogous to deluge/qbittorrent port pinning).

### Implementation Status
Present on current `main` (correct per the per-user lighttpd reverse-proxy pattern, ADR 0009):
- `scripts/lib/lighttpd/proxyFragments.php:61-67` — an `invidious` fragment for **both** the public
  and private paths the issue requested.
- `scripts/lib/lighttpd/userConfigApply.php:69-77` — reads `~/.invidiousPort`; writes the fragment
  iff a valid port is pinned, else removes it. Source comment: *"only when the user or installer pins
  a local port explicitly."*
- `etc/skel/www/index.php` — an Invidious tab definition that renders **only** when the fragment exists.
- `scripts/lib/tests/development/invidiousLighttpdProxyTest.php` + `docs/contracts.md:440` — test + contract.

**Nothing writes `~/.invidiousPort`** (whole-repo: only the reader above), so the wiring is dormant —
no process ever listens behind it. This is the basis for the maintainer's 2026-04-01 "it's only the
proxy config" note, which is accurate.

### What's Missing
Installer; `config.yml`; PostgreSQL; the Invidious **companion**; the enforcement script; anything that
pins the port. Invidious is **not** in the one-click app catalog (`scripts/lib/user/dockerInstallLsio.php`
lists jellyfin/qbittorrent/radarr/sonarr/prowlarr/mariadb/phpmyadmin only), and there is no
LinuxServer-maintained Invidious image (per the 2026-02-01 comment).

### Mission Alignment
- **Honesty / no hallucinations:** the "YouTube blocks datacenter IPs" framing is more nuanced than a
  blanket block, but the sharpened reasoning (no customer-facing residential egress at fleet scale +
  crash-loop-on-block + heavy per-user footprint) still lands on do-not-build, so the verdict is unchanged.
- **Reliability over heroics:** Invidious is a perpetual cat-and-mouse with YouTube; a preconfigured
  fleet feature that breaks on every upstream YouTube change is an anti-reliability feature.
- **Investigate then act:** the remaining action is a decision, not more investigation.

### Devil's Advocate
*Why build it anyway?* The operator wants sovereign tooling over paid vendors, the proxy paths already
exist, and a capable user can self-host Invidious today via the generic per-user Docker + lighttpd
pattern. *Rebuttal:* "the user can already self-host it via the generic pattern" is precisely why the
**bespoke preconfigured installer** is the part that should not exist — and the per-user, fleet-wide,
working-out-of-the-box version (what the issue asks for) is blocked on egress that is not available for
customer video at scale, plus a heavy per-user footprint (PostgreSQL + companion) and ToS/DMCA exposure.

### Anti-Pattern Hunt
- REGEX_FOR_SEMANTICS: **PASS** — no semantic enforcement proposed.
- PREMATURE_OPTIMIZATION: **PASS** — recommendation is do-not-build; nothing optimized prematurely.
- FALSE_COMPLETION_CLAIM: **PASS** — explicitly PARTIAL; no "done" claimed.
- OPERATOR_HANDOFF_ANTIPATTERN: **PASS** — the engineering verdict (SHOULD-NOT-IMPLEMENT) is *made*
  here; only the close/relabel (operator's own feature request) is left to the operator, which is
  genuine business authority, not a cop-out.
- FABRICATED_AUTHORITY_OR_PATH: **PASS** — all file paths verified from current `main`.
- FAIL_OPEN_SAFETY_BEHAVIOR: **PASS** — the wiring fails *closed* (no port → no fragment).
- DRY_RUN_THEATER: **PASS** — n/a.
- INVESTIGATION_LOOP_THEATER: **FLAGGED on the issue, avoided in this pass** — this is ~the 5th-6th
  pass; the loop persists because the decision was never recorded/labeled/closed, not for lack of
  analysis. This report adds verification + a record correction + a decision ask, and recommends no
  further investigation.

### Related Incidents & Data
~10 QA cycles (Apr 1-10 2026) all skipped/failed on this issue; the 2026-05-29 deep investigation; a
2026-06-01 re-confirmation. The very first 2026-04-01 "investigation" was a flagged hallucination. No
customer tickets requesting Invidious were found — low external demand signal.

### Cross-Issue Analysis
- **#378** (exclude `should-not-implement` from the candidate pool) — RELATED systemic fix, but
  **insufficient for #52 alone**: #52 is labeled only `enhancement`, so #378 would not catch it until
  a `should-not-implement` label is applied.
- **#37** (User Settings; disable certain software) — RELATED; the natural home for any enable/enforce toggle.
- **#18** (LinuxServer.io container testing) — RELATED but not blocking (Invidious is not an LSIO image).
- **#171** (container runtime isolation research), **#59** (per-user Docker app + lighttpd proxy
  pattern) — RELATED reuse context. No duplicate, no supersession.

### Dependencies & Blockers
Hard external blocker: Invidious requires reaching YouTube, which blocks datacenter ranges; per-user
customer **video** streaming needs residential-classified egress that is not available at fleet scale.
Loop-termination depends on a maintainer label/ADR/close decision (and, systemically, on #378).

### Reuse, Refactor & Test Strategy
- **Reuse:** the per-user lighttpd `custom.d` reverse-proxy pattern (ADR 0009) already exposes any
  user-run Docker web app over HTTPS — so a bespoke Invidious installer is the part that shouldn't exist.
- **Refactor:** none worth doing *for* this feature; the only positive-value code action is the
  opposite (eventually gating/removing the dormant placeholder — LOW priority, it is harmless and
  doubles as an opt-in hook).
- **Tests (only if ever built):** a real end-to-end "does a video actually play from this host's own
  egress" check — which is exactly the unproven point; that test failing *is* the evidence not to ship.
- **Minimum-code-for-correctness:** **no new code** ("the best part is no part"); record the decision.
- **Mission/business signal:** none can be named truthfully while the feature cannot serve a single
  working video per-user at fleet scale; a crash-looping deploy generates support load, not retention.

### Deep Research Gate
SKIPPED: prior-investigation-count (5-6 prior passes incl. two deep investigations) + internal-evidence-sufficient
(datacenter-IP viability + egress envelope already established and re-verified from source). Further deep
research is investigation-loop theater per this prompt's own skip criteria.

### Architectural Minimization
- **Step A — dumbest requirement:** "Invidious, preconfigured completely, fleet-wide." Written
  2024-02-15 on the then-true premise that self-hosted Invidious works. The mechanism (self-host on
  fleet IPs) became non-viable as YouTube hardened; the requirement's *premise* is the failure point.
- **Step B — alternatives:**
  - DELETE: (1) decline the feature (delete the requirement) — **recommended path**; (2) later
    remove/gate the dormant proxy+tab placeholder — LOW priority (harmless, doubles as opt-in hook).
  - MERGE: fold any future "user-run web app" need into the generic Docker + lighttpd `custom.d`
    pattern (already exists) — makes a bespoke installer redundant.
  - REFACTOR: generalizing the proxy catalog unlocks nothing here — the blocker is egress, not plumbing.
  - NULL: if we don't build it, nothing breaks; users wanting it can self-host via the generic pattern.
    Building it instead risks a crash-loop on a shared host. **DELETE/NULL wins.**
- **Step C:** alternative wins — **no minimal design proposed** (the DELETE/NULL path is correct).

### ADR Adherence
- **ADHERES** ADR 0009 (per-app reverse proxy in per-user lighttpd) — the proxy placement is correct.
- **EXTENDS** ADR 0015 (defer browser console until packaged + reviewed) — same "defer-until-viable,
  no placeholder dead UX" logic. No Invidious ADR exists (latest is 0020) → see RECOMMEND_ADR.

### Final AQFM Sweep
Asked memory the full draft (issue, draft conclusion, evidence files, draft recommendation, request
for contradicting evidence). It retrieved only the issue's own cached entry and surfaced no
contradiction. **Verdict: HELD** on the sources read directly this pass (issue thread, internal
YouTube-landscape research incl. its operator-verified correction, the sovereign-egress reference, and
current-`main` source).

### Adversarial Self-Challenge
- **Strongest finding 1** — the wiring is inert (nothing pins the port). Disproving search: whole-repo
  writer search for the port dotfile → only a reader found. **Held.**
- **Strongest finding 2** — the feature is non-viable as specified. Disproving search 1: looked for
  evidence YouTube unblocks self-hosted Invidious for fleet-scale per-user video → none. Disproving
  search 2 (broader): the only block-dodging path identified internally is not usable for per-user
  video at fleet scale, and per-seedbox-host video viability is unproven. **Held** (with the reasoning
  sharpened from a blanket "datacenter block" to the egress/footprint constraints above).
- **Least-evidenced finding** (lowest citations) — "the loop persists because of labeling + no
  close." Disproving search surfaced a **2026-06-01 re-investigation I had initially missed**, which
  caused me to **revise my draft** from NEEDS_RESEARCH to SHOULD-NOT-IMPLEMENT (the per-host video
  test is genuinely untested but LOW-value — success still leaves the footprint/maintenance/ToS
  downsides). This is the self-challenge working as intended.

### Recommendation
**SHOULD-NOT-IMPLEMENT** (as specified), re-confirmed from current `main`. The honest path is
**decline or defer-until-upstream-viability-returns**, not build. Users wanting Invidious today can
self-host via the generic per-user Docker + lighttpd reverse-proxy pattern (ADR 0009).

**To end the recurring re-investigation, the decision needs to be recorded** — the issue keeps being
re-selected because it carries no decisive label, has no ADR, and is open. Concretely: apply a
`should-not-implement` (or `wontfix`) label and record the ADR below, then close. (#378 is the
systemic selector fix, but it cannot help #52 until the label exists.)

### Additive Recommendation Flags
- **RECOMMEND_ADR** — *"Decline/defer preconfigured Invidious — datacenter-IP / egress non-viability."*
  Context: #52 requested per-user preconfigured Invidious; it depends on YouTube, which blocks
  datacenter ranges; per-user customer video needs residential-classified egress not available at
  fleet scale; a blocked instance crash-loops on a shared host; only inert proxy/UI plumbing exists.
  Options: build now (rejected — ships a crash-looper); BYO-residential-proxy one-click (rejected — not
  "preconfigured," still crash-loops by default); **defer/decline (chosen)**. Consequence: recording
  this ends the re-investigation loop; the dormant placeholder may be gated/removed later (LOW
  priority). Mirrors ADR 0015.

### Auto-Close Decision
AUTO_CLOSED_BLOCKED: trigger-not-met; reason=rationale-not-supersession. SHOULD-NOT-IMPLEMENT here is
upstream/egress **non-viability**, not supersession (no open successor subsumes #52); the issue was
operator-authored and the operator wanted it built, so the close/relabel is the maintainer's call. I
will not close it unilaterally despite the loop cost.

---
Väinämöinen <noreply@pulsedmedia.com>
*First the origin; then the cure. The plumbing is sound; the upstream is not — yet. This thread needs a recorded decision, not a sixth investigation.*

**Comment by MagnaCapax** [comment]
## #52 — convergent decline re-confirmed (4th pass); agent investigation halting here

This is the **fourth independent investigation** to reach the same verdict (see the 2026‑05‑29, 2026‑06‑01, and 2026‑06‑02 reports above), on top of ~10 earlier QA cycles. The decisive facts were re-verified against current `main` this pass and **nothing has changed**, so — per our own anti‑loop discipline — this note deliberately does **not** re-derive the technical case a fourth time.

**Verdict (unchanged): SHOULD‑NOT‑IMPLEMENT as specified.**
The per-user reverse-proxy + tab scaffolding for both the public (`/public-USERNAME/invidious`) and private (`/user-USERNAME/apps/invidious`) paths exists but is **inert** — nothing pins the local port, so the proxy is never written and the tab never renders. There is no installer, config, database, companion, or enforcement script, and Invidious is not in the one-click container catalog. A self-hosted Invidious cannot reliably serve YouTube from datacenter-class IPs in 2026 (per Invidious's own docs), crash-loops by default on a shared host, and is a perpetual cat-and-mouse to keep working — the opposite of "preconfigured completely." This declines *the feature as specified*, not sovereign YouTube tooling in general.

**Why this keeps reappearing:** the decision was never recorded and the issue was never closed/relabeled — not a lack of investigation. It carries only the `enhancement` label, so the selector fix tracked in #378 cannot exclude it until a decisive label is applied. The remaining step is a maintainer call, not another agent pass:

- close as won't-do (as specified), **or**
- relabel/defer and record the ADR drafted in the comments above.

Not auto-closed here: this is a viability decline (not a supersession), and you authored the feature — so the close/relabel is yours to make.

— Väinämöinen
*First the origin; then the cure.*

**Comment by MagnaCapax** [comment]
## Applying `blocked` — pausing automated re-investigation, pending a maintainer decision

The investigation passes above have converged, and re-verifying against current `main` this pass changed nothing:

- The per-user reverse-proxy + app-tab scaffolding for both the public (`/public-USERNAME/invidious`) and private (`/user-USERNAME/apps/invidious`) paths exists, but it is **inert** — nothing pins the local port, so the proxy fragment is never written and the tab never renders. There is no installer, config, database, companion, or enforcement script.
- A self-hosted Invidious cannot reliably serve YouTube from datacenter-class IPs in 2026 (per Invidious's own docs); the default failure mode on a blocked IP is a crash loop, and keeping it working is a continuous cat-and-mouse — the opposite of "preconfigured completely."

This issue keeps resurfacing not for lack of investigation but because the decision was never recorded, so the automated pipeline keeps re-selecting it. I'm applying **`blocked`** to stop that churn. This is a **status, not a verdict** — it means the issue can't proceed without a maintainer decision, not that the feature is rejected.

The call is yours, since you authored it and the sovereign-tooling intent is sound:
- **remove `blocked`** to return it to the pipeline,
- **`should-not-implement` / close** to record a decline (as specified), or
- **rescope** to a smaller feature (e.g. a one-click installer where the user supplies their own residential egress).

— Väinämöinen
*First the origin; then the cure.*
